src/hiveboard_bridge/src/main.cpp

```diff
@@ -20,29 +31,19 @@ int main(int argc, char** argv) {
     ros::init(argc, argv, "hiveboard_bridge");
 
     int port = getTcpServerPort();
-    TCPServer socket(port);
+    TCPServer tcpServer(port);
 
     ROS_INFO("Listening for incoming clients...");
-    socket.listen();
+    tcpServer.listen();
     ROS_INFO("Client connected.");
 
-    char sendValue[] = "Hello world from ROS!";
-    socket.send(sendValue, strlen(sendValue));
-    ROS_INFO("Sent a message from server to client");
+    HiveMindHostDeserializer deserializer(tcpServer);
+    MessageHandler messageHandler;
 
-    char buf[10] = "";
-    char okBuf[3] = "Ok";
-    int i = 0;
-    while (true) {
-        ROS_INFO("%d", i++);
+    ReceiveThreadAction receiveThreadAction(deserializer, messageHandler);
 
-        bzero(buf, 10);
-        socket.read(buf, 10, true);
-        ROS_INFO("Data: %s", buf);
-        socket.send(okBuf, 3);
-    }
+    StreamListener streamListener(receiveThreadAction);
```

Comment:
Wouldn't i be easier to replace the "streamlistener" thread to just a while loop with ros::spin_once?

Reply:
Perhaps. I'm not sure yet how I'm going to manage the two-way communication with the callbacks. I might remove this in a not-so-distant future if we realise it is not necessary.